bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirHandler.java

```diff
@@ -0,0 +1,292 @@
+                    logger.warn("Unhandled command type: {}", command.toString());
+                }
+            } catch (final RuntimeException e) {
+                logger.warn("Updating ComfoAir failed: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        ComfoAirConfiguration config = getConfigAs(ComfoAirConfiguration.class);
+        String serialPort = config.serialPort;
+
+        if (StringUtils.isNotEmpty(serialPort)) {
+            comfoAirConnector = new ComfoAirSerialConnector(serialPortManager, serialPort, BAUDRATE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            return;
+        }
+        if (comfoAirConnector != null) {
+            comfoAirConnector.open();
+            updateStatus(ThingStatus.ONLINE);
+
+            List<Channel> channels = this.thing.getChannels();
+
+            poller = scheduler.scheduleWithFixedDelay(() -> {
+                for (Channel channel : channels) {
+                    updateChannelState(channel);
+                }
+            }, 0, (config.refreshInterval > 0) ? config.refreshInterval : DEFAULT_REFRESH_INTERVAL, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (comfoAirConnector != null) {
+            comfoAirConnector.close();
+        }
+
+        final ScheduledFuture<?> localPoller = poller;
+
+        if (localPoller != null && !localPoller.isCancelled()) {
+            localPoller.cancel(true);
+            poller = null;
+        }
+    }
+
+    private void updateChannelState(Channel channel) {
+        try {
+            if (!isLinked(channel.getUID())) {
+                return;
+            }
+            String commandKey = channel.getUID().getId();
+            ComfoAirCommand readCommand = ComfoAirCommandType.getReadCommand(commandKey);
+            State state = sendCommand(readCommand, commandKey);
+
+            updateState(channel.getUID(), state);
+        } catch (IllegalArgumentException e) {
+            logger.warn("Unknown channel {}", channel.getUID().getId());
+        }
+    }
+
+    private State sendCommand(ComfoAirCommand command, String commandKey) {
+        ComfoAirSerialConnector comfoAirConnector = this.comfoAirConnector;
+
+        if (comfoAirConnector != null) {
+            Integer requestCmd = command.getRequestCmd();
+            Integer replyCmd = command.getReplyCmd();
+            int[] requestData = command.getRequestData();
+
+            Integer preRequestCmd;
+            Integer preReplyCmd;
+            int[] preResponse = new int[0];
+
+            if (requestCmd != null) {
+                switch (requestCmd) {
+                    case 0x9f:
```

Comment:
Would it be possible to give these constants names? Or do they not really have names to begin with?